src/devol.cpp

```diff
@@ -103,6 +107,18 @@ void devol(double VTR, double f_weight, double f_cross, int i_bs_flag,
             t_tmpP = ta_oldP.col(i);            // t_tmpP is the vector to mutate and eventually select
 
             permute(ia_urn2.memptr(), urn_depth, i_NP, i, ia_urntmp.memptr()); // Pick 4 random and distinct 
+            
+            //Trigger JADE algorithm when d_c <> 0 (randomize cross-over and weight coefficient)
+            if(d_c>0){
+              f_cross = Rcpp::as<double>(rnorm(1, mean_cross, 0.1));
+              f_cross = f_cross > 1.0 ? 1 : f_cross;
+              f_cross = f_cross < 0.0 ? 0 : f_cross;
+              do{
+                f_weight = Rcpp::as<double>(rcauchy(1, mean_weight, 0.1));
+                f_weight = f_weight > 1.0 ? 1.0 : f_weight;
+              } while (f_weight <= 0.0);
+            }
+            
```

Comment:
I think I know the reason. `f_cross` and `f_weight` are passed in as const parameters, the code above attempts to change them. let me make the change